### DISCUSSION TITLE
SVG to WPF Control

### DISCUSSION THREAD

**Discussion by ppberube** [dotnetprojects/SVGImage#92] [General]
Hi,

Looking for an experimented programmer who could work on the following project : we don't have any experience in this so we thought this forum would be a good starting point.

- A designer would provide a SVG of a circular gauge and a linear gauge
- The task would be to generate WPF controls based on the SVGs
- The WPF controls would be used in our in house application, used to display field-values in the automotive industry.

For example, we use similar controls as the following : https://docs.devexpress.com/WPF/115093/controls-and-libraries/gauge-controls
and we would like the same approach.

Please contact me if you are interrested and would like to know more !

**Comment by paulushub** [comment]
@ppberube How to contact you?

**Comment by ReIaxo** [comment]
Hi. I have a similar requirement. I want to build a Control/Image library, where the SVGs displayed change depending on "Input" Properties of the Control. The main usage will be to show the state of valves, motors and other actors and the connecton between them. Maybe indicated by changeing the color of a path in the svg or its thickness?

Does someone know if something like this already exists or which will be the best approach to create it?

This would be one single control which can be connected to others. 
![grafik](https://github.com/dotnetprojects/SVGImage/assets/44195536/780ace7a-fa09-4c96-8aff-5ea4e463d689)

**Reply by paulushub** [reply]
1. Determine which library meets your rendering requirements: `SVGImage` or `SharpVectors`.
2. Then consider the available features of each library (if both meet your needs)

### SVGImage
- `SVGImage` parses the SVG into its own implementation of shapes.
- You can access these shapes and animate according to your needs
- There is no hit-test support in the controls, you will have to implement it if you need user interactivity.

### SharpVectors
- `SharpVectors` parses SVG into an extension of .NET `XmlDocument` class, so all the navigations, such as XPath queries, are still available.
- There is an implementation class `WpfDrawingDocument` which saves and provide a link between the SVG elements and the  corresponding rendered `Drawing` objects.
- There is also an implementation of hit-test, which so far is working well. WPF does not provide hit-test for `Drawing` objects, only in visual objects.
- See hit-testing: https://github.com/ElinamLLC/SharpVectors/issues/148
- See a user-implementation using the hit-testing: https://github.com/ElinamLLC/SharpVectors/issues/106
